metricflow/plan_conversion/node_processor.py

```diff
@@ -297,7 +297,7 @@ def add_multi_hop_joins(
 
         return list(x.node_with_multi_hop_elements for x in all_multi_hop_join_candidates) + list(nodes)
 
-    def remove_unnecessary_nodes(
+    def heuristically_prune_nodes(
```

Comment:
What's the reason for the rename? Seems harder to understand the new name 😅

Reply:
Yeah, that would definitely benefit from a separate commit with a commit message explaining it.

Reply:
I thought it would be more helpful, but either works for me. Let me revert it.